dotnet/src/Agents/Abstractions/Definition/AgentDefinition.cs

```diff
@@ -14,171 +14,60 @@ public sealed class AgentDefinition
     /// Gets or sets the id of the deployed agent.
     /// </summary>
-    public string? Id
-    {
-        get => this._id;
-        set
-        {
-            Verify.NotNull(value);
-            this._id = value;
-        }
-    }
+    public string? Id { get; set; }
 
     /// <summary>
     /// Gets or sets the type of the  agent.
     /// </summary>
-    public string? Type
-    {
-        get => this._type;
-        set
-        {
-            Verify.NotNull(value);
-            this._type = value;
-        }
-    }
+    public string? Type { get; set; }
 
     /// <summary>
     /// Gets or sets the name of the  agent.
     /// </summary>
-    public string? Name
-    {
-        get => this._name;
-        set
-        {
-            Verify.NotNull(value);
-            this._name = value;
-        }
-    }
+    public string? Name { get; set; }
 
     /// <summary>
     /// Gets or sets the description of the agent.
     /// </summary>
-    public string? Description
-    {
-        get => this._description;
-        set
-        {
-            Verify.NotNull(value);
-            this._description = value;
-        }
-    }
+    public string? Description { get; set; }
 
     /// <summary>
     /// Gets or sets the instructions for the agent to use.
     /// </summary>
-    public string? Instructions
-    {
-        get => this._instructions;
-        set
-        {
-            Verify.NotNull(value);
-            this._instructions = value;
-        }
-    }
+    public string? Instructions { get; set; }
 
     /// <summary>
     /// Gets or sets the metadata associated with the agent.
     /// </summary>
-    public IDictionary<string, object?>? Metadata
-    {
-        get => this._metadata;
-        set
-        {
-            Verify.NotNull(value);
-            this._metadata = value;
-        }
-    }
+    public IDictionary<string, object?>? Metadata { get; set; }
 
     /// <summary>
     /// Gets or sets the model used by the agent.
     /// </summary>
-    public ModelDefinition? Model
-    {
-        get => this._model;
-        set
-        {
-            Verify.NotNull(value);
-            this._model = value;
-        }
-    }
+    public ModelDefinition? Model { get; set; }
 
     /// <summary>
     /// Gets or sets the collection of input variables used by the agent.
     /// </summary>
-    public IList<InputVariable> Inputs
-    {
-        get => this._inputs ??= [];
-        set
-        {
-            Verify.NotNull(value);
-            this._inputs = value;
-        }
-    }
+    public IList<InputVariable>? Inputs { get; set; }
 
     /// <summary>
     /// Gets or sets the collection of output variables supported by the agent.
     /// </summary>
-    public IList<OutputVariable> Outputs
-    {
-        get => this._outputs ??= [];
-        set
-        {
-            Verify.NotNull(value);
-            this._outputs = value;
-        }
-    }
+    public IList<OutputVariable>? Outputs { get; set; }
```

Comment:
Is intended to non serialize this property when `null`, if not, this keeps the behavior of the backfield implementation (apply for other `IList<...>`)

```suggestion
    public IList<OutputVariable> Outputs { get; set; } = [];
```